### ISSUE TITLE
Provide file-system-based discovery of synced folder root

### ISSUE THREAD

**Issue by siegel** [SamSchott/maestral#546]
**Is your feature request related to a problem? Please describe.**

Some applications (such as [BBEdit](https://www.barebones.com/products/bbedit/)) provide built-in support for recognizing that Dropbox is in use, and supporting syncing of user configuration data thereby.

Maestral evidently does not do this, so although it is a Dropbox work-alike, customers who are using only Maestral with their Dropbox accounts don't get this benefit.

**Describe the solution you'd like**

In BBEdit, Dropbox is detected by looking for `~/.dropbox/info.json` and loading the JSON file, which typically looks something like:

```json
{
	"personal": {
		"path": "/Users/siegel/Dropbox",
		"host": 12345678,
		"is_team": false,
		"subscription_type": "Pro"
	},
}
```

If this file exists,  BBEdit will fetch the `personal` dictionary; and if that is found, `personal.path` is used as a fully qualified path to the local synced directory root. If the `personal` dictionary is not present, but the `business` dictionary is, BBEdit will use `business.path` instead. None of the other keys matter (to BBEdit).

It would be great if Maestral supported this on-disk format as well; if it did, then it would be a completely transparent replacement for Dropbox in this respect, without requiring changes to BBEdit (or any other client apps which use the same mechanism to find the local Dropbox root).

This is a documented and supported mechanism: Dropbox's own documentation is available here: [https://help.dropbox.com/installs-integrations/desktop/locate-dropbox-folder](https://help.dropbox.com/installs-integrations/desktop/locate-dropbox-folder).

**Comment by siegel** [comment]
_bump_

**Comment by samschott** [comment]
Apologies for not responding to this earlier. I can see how this would be beneficial but there also a number of gotchas which I'll need to think about:

1. Were the first–party Dropbox client is is still used in addition to Maestral or was used in the past, the json file will already exist. We cannot guess the users intention and just replace this file. A dialog during setup in such cases could resolve that.

2. The first-party clients supports syncing only two account, and only if they are linked Business and Personal accounts. This is contrary to Maestral's model of not limiting the number of accounts, meaning that we could not meaningfully populate the json file in some cases while still following Dropbox's spec. This should be made clear to the user.

**Comment by siegel** [comment]
> Were the first–party Dropbox client is is still used in addition to Maestral or was used in the past, the json file will already exist.
I agree that it would not be a good idea to overwrite the file; but if the Dropbox application is not running, and the file matches the user's Maestral configuration, then leaving it alone seems appropriate. Otherwise, yes, prompting the user is probably the only reasonable course of action>

> The first-party clients supports syncing only two account, and only if they are linked Business and Personal accounts. This is contrary to Maestral's model of not limiting the number of accounts, meaning that we could not meaningfully populate the json file in some cases while still following Dropbox's spec.

Although Dropbox's example shows the JSON file containing keys for "personal" and "business", I am quite sure there's no reason you couldn't use other key names for other accounts. However, since the documentation only shows "personal" and "business", if you only write those, that will suffice.

**Comment by samschott** [comment]
Sure. My concern was about situations for example with two "personal" accounts. Which account should Maestral actually key with "personal" for BBEdit to pick up? There are again multiple options, such as "first come, first served" or a user dialog.

The issue which I see with dialogs is that they will need to explain a somewhat complicated situation:

1. The purpose of `~/.dropbox/info.json` and how it is used.
2. Subtleties around multiple accounts if applicable.

Automatically doing the "right" / "best" thing may however be annoying to the user.

Independent of UI questions, Maestral would be piggybacking on a Dropbox-specific file and I'm not sure if I like that idea. If users already know what this file does and how its used by 3rd party apps, they might as well create / edit it manually.

**Comment by siegel** [comment]
> My concern was about situations for example with two "personal" accounts.

How frequently does that use case occur for people using Maestral?

> Maestral would be piggybacking on a Dropbox-specific file and I'm not sure if I like that idea. 

I mean... you're already piggybacking on their API. :-)

> If users already know what this file does and how its used by 3rd party apps, they might as well create / edit it manually.

That predicate doesn't work, unfortunately. Our experience has been that customers *don't* know what the file does (or even, necessarily, that it exists), and it's JSON which is not exactly the simplest thing in the world to edit by hand. (In our experience, customers using Dropbox — or even Maestral — are not necessarily more technically adept than any other macOS user, nor significantly more likely to dig into the nuts and bolts. They just want the thing to work, and when it doesn't we're the ones they write to.)
